numba/tests/test_parfors.py

```diff
@@ -1838,6 +1840,22 @@ def test_impl(x):
         x = np.ones((2, 2, 2, 2, 2, 15))
         self.check(test_impl, x)
 
+    def test_global_loop_nest_stop(self):
+        # issue7213
+        def test_impl():
+            a = np.empty((2,))
+            for k in prange(2):
+                b = np.zeros((_GLOBAL_INT_FOR_TESTING2,
+                              _GLOBAL_INT_FOR_TESTING2))
+                c = np.zeros((_GLOBAL_INT_FOR_TESTING2,
+                              _GLOBAL_INT_FOR_TESTING2), dtype=np.complex128)
+                d = b * c
+                e = np.diag(_GLOBAL_ISSUE7213_ARR + d)
+                a[k] = 1
+            return a
+
+        self.check(test_impl)
+
```

Comment:
I'm struggling to get this to reproduce the original issue. On current [HEAD](https://github.com/numba/numba/tree/5baa1ce53073a786060078a8cbe0ee57b1fb67d2) of mainline this test code seems to work:
```python
$ git rev-parse --short HEAD
5baa1ce

$ cat test_7243.py 
from numba import njit, prange
import numpy as np

_GLOBAL_INT_FOR_TESTING1 = 17
_GLOBAL_INT_FOR_TESTING2 = 5
_GLOBAL_ISSUE7213_ARR = np.ones((_GLOBAL_INT_FOR_TESTING2,
                                 _GLOBAL_INT_FOR_TESTING2)).astype(np.complex128)

@njit(parallel=True)
def test_impl():
    a = np.empty((2,))
    for k in prange(2):
        b = np.zeros((_GLOBAL_INT_FOR_TESTING2,
                        _GLOBAL_INT_FOR_TESTING2))
        c = np.zeros((_GLOBAL_INT_FOR_TESTING2,
                        _GLOBAL_INT_FOR_TESTING2), dtype=np.complex128)
        d = b * c
        e = np.diag(_GLOBAL_ISSUE7213_ARR + d)
        a[k] = 1
    return a

print(test_impl())

$ python test_7243.py
[1. 1.]

```

Reply:
@stuartarchibald I also cannot replicate the issue with the latest Numba.  I went back to a July version of Numba and it also didn't replicate there at the time when this issue was originated.  I hesitate to say abandon this PR unless we could prove that the circumstance that this PR is designed to fix can't happen.  I don't know how to force the situation to occur at the moment either.  Suggestions?